stable/redis/templates/redis-slave-deployment.yaml

```diff
@@ -71,6 +69,8 @@ spec:
         image: {{ template "redis.image" . }}
         imagePullPolicy: {{ .Values.image.pullPolicy | default "" | quote }}
 {{- $command := default .Values.master.command .Values.slave.command }}
+        {{- /* Include master securityContext if slave securityContext not defined */ -}}
+        {{ include "redis.slave.securityContext" . | indent 8 }}
```

Comment:
This would also add the `fsGroup` directive inside the container section so it wont work. 
You should do the same you did for the master. Split `runAsUser` and `fsGroup` configuration in container and pod sections respectively.